benches/throughput.rs

```diff
@@ -132,16 +139,16 @@ fn initialize_environment(
             let _ = tokio::join!(app.wait_termination(), echo.run());
         });
         let mut hbone = ta
-            .socks5_connect(helpers::with_ip(
-                echo_addr,
-                TEST_WORKLOAD_HBONE.parse().unwrap(),
-            ))
+            .socks5_connect(
```

Comment:
This section is repeated often.

e.g.

```
 let mut s =
    e.ta.socks5_connect(helpers::with_ip(
        e.echo_addr,
        TEST_WORKLOAD_TCP.parse().unwrap(),
    ))
    .await;
```

Wdyt about abstracting that logic into a helper function, maybe something called `newTCPStream` or similar. Have it maybe take in the `TEST_WORKLOAD_<SOURCE/HBONE/TCP>` string value as an argument. Would help to maintain that DRY principle while also making it convenient for the next person to extend this test with more benches?

Reply:
I'll think about this. The whole echo server setup is clever but also confusing. You can access the echo server through any ip address on `lo`, but if you want to use ztunnel/socks5, the address you pick to access the echo server needs to be a registered workload.

Point is, I feel like the way its abstracted right now is already confusing, so I didn't want to abstract it any further. I do think there is probably a better way to do this. My first idea it to make `TEST_WORKLOAD_*` actual IP address instead of a string that can be parsed into an ip address. Thoughts?

Reply:
The abstraction I refer to is to really help reduce the repetitive and confusing code in the tests. Using a helper function should actually make it more understandable.